src/main/java/org/isf/menu/gui/LogoutEventListener.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Open Hospital (www.open-hospital.org)
+ * Copyright © 2006-2021 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+ *
+ * Open Hospital is a free and open source software for healthcare data management.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isf.menu.gui;
+
+import java.awt.AWTEvent;
+import java.awt.event.AWTEventListener;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseEvent;
+import java.util.Arrays;
+
+import org.isf.generaldata.GeneralData;
+import org.isf.menu.model.User;
+import org.isf.utils.jobjects.DelayTimerCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LogoutEventListener implements AWTEventListener, Login.LoginListener, DelayTimerCallback {
+
+	@Override
+	public void eventDispatched(AWTEvent e) {
```

Comment:
This seems to be very resource intensive.   Would it be better to do something with the database connection?  Almost every window/screen/dialog hits the database at some point.  Granted it would not be as fine grained as this but it would be a good approximation.    Just a thought ....

Reply:
I agree with you that this solution consumes a lot of resources. Initially I restarted the timer when mouse_enter was done on some element of the screen. It would be necessary to leave the reset timer feature when the user uses the keyboard. 
I am wondering if Open Hospital has pages on which the user could not enter data or move the mouse.